### PR TITLE
updating docs, examples, and adding error check in add()

### PR DESCRIPTION
**Pull request by afshin**

### CODE REVIEW

File: src/commandpalette/palette.ts [hunk 1]
```diff
@@ -878,29 +888,30 @@ class CommandPalette extends Widget implements ICommandPalette {
...
-        return section.title === newCategory;
-      });
-      if (sectionIndex === -1) {
-        // If a section with this header does not exist, add a new section.
-        this._sections.push({ title: item.category, registrations: [item.id] });
-      } else {
-        // If a section exists with this header, add to it.
-        this._sections[sectionIndex].registrations.push(item.id);
-      }
+    // If the item does not need to be shifted into a different section, return;
```
Comment (blink1073) [inline src/commandpalette/palette.ts]: `;` -> `.`

File: src/commandpalette/palette.ts [hunk 2]
```diff
@@ -871,6 +874,13 @@ class CommandPalette extends Widget implements ICommandPalette {
     }));
   }
 
+  /**
+   * Update the local command item by referencing its underlying `Command`.
+   *
+   * @param id - The palette regitration ID for the command item.
```
Comment (blink1073) [inline src/commandpalette/palette.ts]: (sp.)

File: src/commandpalette/palette.ts (hunk 1 above)
Comment (afshin) [inline src/commandpalette/palette.ts]: Ha!

**Comment by blink1073** [comment]
Simpler === Better

**Pull request merged**